### PR TITLE
Update developer guide for Logic and DeleteCommand

### LINKED ISSUE #47: Update Developer Guide
On Architecture Diagrams and EditCommands implementations.

### PR DESCRIPTION
**Pull request by laurenlhy**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -29,6 +46,55 @@ Refer to the guide [_Setting up and getting started_](SettingUp.md).
...
+
+* Cascading of Delete commands has been implemented with the assumption that when a customer
+or order is deleted, all information related to that customer or order must also be removed from CHIM.
+* The cascading of Delete commands is implemented only in one direction (Customer to Order to Cheese).
+Deleting an order will not delete the customer who placed the order.
+Furthermore, deleting a cheese which has been assigned to an order is not allowed.
+This is to prevent any extra erroneous deletions.
+* All `execute()` calls by `DeleteCustomerCommand`, `DeleteOrderCommand` and `DeleteCheeseCommand`
+will call on `Model.AddressBook` which will handle the cascading of delete commands in one place,
+as opposed to implementation within the different `Command` classes.
```
Comment (weixue123) [inline docs/DeveloperGuide.md]: The last line sounds a little confusing, is it possible to rephrase it? (the "as opposed to implementation within the different `Command` classes" part). or we can simply omit it if it's not important?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -21,6 +21,23 @@ Refer to the guide [_Setting up and getting started_](SettingUp.md).
...
+1. `Logic` uses the `AddressBookParser` class to parse the user command.
+2. This results in a `Command` object which is executed by the `LogicManager`.
+3. The command execution can affect the `Model` (e.g. by adding customers, orders or cheeses).
+4. The result of the command execution is encapsulated as a `CommandResult`
+object which is passed back to the `Ui`.
+5. In addition, the `CommandResult` object can also instruct the `Ui` to perform
+certain actions, such as displaying help to the user.
+
+Given below is the Sequence Diagram for interactions within the `Logic` component
+for the `execute("deletecheese 1")` API call.
```
Comment (weixue123) [inline docs/DeveloperGuide.md]: Trailing white space

**Review by weixue123 — APPROVED** [review]
Ignore what I commented earlier about uploading the PUML files (since I think as a group we are not doing that).

Overall lgtm, can merge after making the minor changes.

**Comment by daniellau88** [comment]
Should the sequence diagram specify the arguments?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -29,6 +46,55 @@ Refer to the guide [_Setting up and getting started_](SettingUp.md).
 
 ## **Implementation** [In Progress]
```
Comment (weixue123) [inline docs/DeveloperGuide.md]: Add --- divider and remove [In Progress]

**Pull request merged**